Source/JavaScriptCore/bytecompiler/BytecodeGenerator.h

```diff
@@ -846,8 +845,14 @@ namespace JSC {
 
         ResolveType resolveType();
         RegisterID* emitResolveConstantLocal(RegisterID* dst, const Variable&);
+        RegisterID* emitResolveScopeWithVarKindScope(const Variable&);
+        RegisterID* emitResolveScopeHelper(RegisterID* dst, const Variable&);
         RegisterID* emitResolveScope(RegisterID* dst, const Variable&);
+        RegisterID* emitGetFromScopeWithVarKindStack(RegisterID* dst, const Variable&);
+        RegisterID* emitGetFromScopeWithVarKindDirectArgument(RegisterID* dst, RegisterID* scope, const Variable&);
+        RegisterID* emitGetFromScopeHelper(RegisterID* dst, RegisterID* scope, const Variable&, ResolveMode);
         RegisterID* emitGetFromScope(RegisterID* dst, RegisterID* scope, const Variable&, ResolveMode);
+        RegisterID* emitResolveAndGetFromScope(RegisterID* dst, RegisterID* resolvedScope, const Variable&, ResolveMode, bool isPrivate = false);
```

Comment:
I don't see a reason why we need a resolvedScope parameter here. The one call site that's passing something other than nullptr is looks wrong to me. I think it's actually better if the contract of this API is that we don't define a scope, so we can remove the operand from the bytecode

Reply:
Yes, we need `resolvedScope` for handling non-ResolveAndGetFromScope cases introduced in `emitResolveAndGetFromScope`.